src/full_node/full_node_api.py

```diff
@@ -196,19 +196,27 @@ async def request_proof_of_weight(self, request: full_node_protocol.RequestProof
         return Message("respond_proof_of_weight", full_node_protocol.RespondProofOfWeight(wp, request.tip))
 
     @api_request
-    async def respond_proof_of_weight(self, response: full_node_protocol.RespondProofOfWeight) -> Optional[Message]:
+    @peer_required
+    async def respond_proof_of_weight(
+        self,
+        response: full_node_protocol.RespondProofOfWeight,
+        peer: ws.WSChiaConnection,
+    ) -> Optional[Message]:
+        self.full_node.pow_pending.remove(peer.peer_node_id)
         validated, fork_point = self.full_node.weight_proof_handler.validate_weight_proof(response.wp)
-        if validated is True:
-            # get tip params
-            tip_weight = response.wp.recent_chain_data[-1].reward_chain_sub_block.weight
-            tip_height = response.wp.recent_chain_data[-1].reward_chain_sub_block.sub_block_height
-            self.full_node.sync_store.add_potential_peak(response.tip, tip_height, tip_weight)
-            self.full_node.sync_store.add_potential_fork_point(response.tip, fork_point)
-            return Message(
-                "request_sub_block",
-                full_node_protocol.RequestSubBlock(uint32(tip_height), True),
-            )
-        return None
+        if not validated:
+            self.log.warning("bad weight proof, disconnecting peer")
```

Comment:
You can just raise an exeption instead of peer.close, and the server will do the closing automatically. I guess this way works too though

Reply:
we should raise an exception so it behaves the same way as everything else